docs/wiki/The library - Module design.md

```diff
@@ -391,6 +391,8 @@ Within a bicep file, use the following conventions:
   - `Conditional` - The parameter value can be optional or required based on a condition, mostly based on the value provided to other parameters.
   - `Optional` - The parameter value is not mandatory. The module provides a default value for the parameter.
   - `Generated` - The parameter value is generated within the module and should not be specified as input.
+- If parameters map to resource properties, they should not contain the resource's name as it would introduce redundancy. For example, the `name` property of the Key Vault module should be just that and not `keyVaultName`. The rationale is that the consumers know that the name is for the Key Vault if they deploy its module.
+- Further, if a property value allows a single value only, there is no need to introduce a parameter for it. Instead it can be hardcoded into the deployment. For example, the name of a blobServices resource can only be `default`. Hence we can implement its name property directly as `name: 'default'`.
```

Comment:
Far from easy, but should we update the paragraph to include the updated approach i.e., using the variable name and make the diagnosticName empty?

Reply:
I'm also thinking that using a variable for the hardcoded value even when diagnostic is not supported would be easier to describe and provide guidelines for. I sense you won't agree, passing you the ball😜

Reply:
I definitely wouldn't agree doing it 😉 but yeah, describing it would be easier 😄
All add a "special case" entry here. I hope thought that we won't end up with too many in the long term. Already many contributions seem to ignore the existing ones

Reply:
Done

Reply:
For 'all' it will anyways become quite interesting as I believe for some resources we already apply some logic to the name which we now need to intertwine with this.

Reply:
So I'm not sure how to update the description for that. Or do you already have something in mind?
I can also update the other diagnostic settings in this PR if you want :)

Reply:
Let's keep them separated as this PR is done already (will just have a final review). How to update the wiki will become more clear once we update others. This is correct for the current implementation so I think is good